packages/runtime/src/widgets/Avatar/Avatar.tsx

```diff
@@ -108,20 +110,24 @@ export const Avatar: React.FC<AvatarProps> = (props) => {
           onClose={handleMenuClose}
           open={isMenuOpen}
         >
-          {values?.menu.map((menuItem, idx) => (
-            <MenuItem key={idx} onClick={(): void => handleMenuClick(menuItem)}>
-              {menuItem.icon ? (
-                <ListItemIcon>
-                  <Icon
-                    color={menuItem.icon.color}
-                    name={menuItem.icon.name}
-                    size={menuItem.icon.size}
-                  />
-                </ListItemIcon>
-              ) : null}
-              {menuItem.label}
-            </MenuItem>
-          ))}
+          {values?.menu.map((menuItem, idx) => {
+            const icon = isString(menuItem.icon)
+              ? { name: menuItem.icon }
+              : menuItem.icon;
```

Comment:
This can be a common util to convert either a string or object props to an icon component.